### PR TITLE
perf: improve redis load in the event persisters

### PR DESCRIPTION
**Pull request by cre8ivejp**
Instead of requesting the same data for each event, I changed it to request just once per batch. This will improve the load on Redis.

### CODE REVIEW

File: pkg/eventpersisterops/persister/evaluation_event.go [hunk 1]
```diff
@@ -63,20 +63,30 @@ func NewEvalUserCountUpdater(
...
+			u.logger.Error("failed to list auto ops rules",
+				zap.Error(err),
+				zap.String("environmentNamespace", environmentNamespace),
+			)
+			handledCounter.WithLabelValues(codeFailedToListAutoOpsRules).Inc()
+			// Make sure to retry all the events in the next pulling
+			for id := range events {
+				fails[id] = true
+			}
+			return fails
```
Comment (masaaania) [inline pkg/eventpersisterops/persister/evaluation_event.go:77]: If the `encironmentEventMap` contains the two environments' events and some error occurs when executing `listAutoOpsRules` on the processing of the first environment, I assume this `fails` doesn't contain the eventIDs of the second environment. Is it OK?

File: pkg/eventpersisterops/persister/goal_event.go [hunk 2]
```diff
@@ -70,14 +70,30 @@ func NewGoalUserCountUpdater(
...
+			u.logger.Error("failed to list auto ops rules",
+				zap.Error(err),
+				zap.String("environmentNamespace", environmentNamespace),
+			)
+			handledCounter.WithLabelValues(codeFailedToListAutoOpsRules).Inc()
+			// Make sure to retry all the events in the next pulling
+			for id := range events {
+				fails[id] = true
+			}
+			return fails
```
Comment (masaaania) [inline pkg/eventpersisterops/persister/goal_event.go:84]: same here.

File: pkg/eventpersisterdwh/persister/goal_event.go [hunk 3]
```diff
@@ -89,21 +89,30 @@ func (w *goalEvtWriter) Write(
...
+			w.logger.Error("failed to list experiments",
+				zap.Error(err),
+				zap.String("environmentNamespace", environmentNamespace),
+			)
+			handledCounter.WithLabelValues(codeFailedToListExperiments).Inc()
+			// Make sure to retry all the events in the next pulling
+			for id := range events {
+				fails[id] = true
+			}
+			return fails
```
Comment (masaaania) [inline pkg/eventpersisterdwh/persister/goal_event.go:103]: same here.

File: pkg/eventpersisterdwh/persister/evaluation_event.go [hunk 4]
```diff
@@ -83,23 +83,34 @@ func (w *evalEvtWriter) Write(
...
+			w.logger.Error("failed to list experiments",
+				zap.Error(err),
+				zap.String("environmentNamespace", environmentNamespace),
+			)
+			handledCounter.WithLabelValues(codeFailedToListExperiments).Inc()
+			// Make sure to retry all the events in the next pulling
+			for id := range events {
+				fails[id] = true
+			}
+			return fails
```
Comment (masaaania) [inline pkg/eventpersisterdwh/persister/evaluation_event.go:97]: same here.

File: pkg/eventpersisterops/persister/evaluation_event.go (hunk 1 above)
Comment (cre8ivejp) [reply pkg/eventpersisterops/persister/evaluation_event.go]: Thanks for pointing that out.
I have fixed the condition.

**Review by masaaania — APPROVED** [review]
Thank you!!

**Pull request merged**